src/lib.rs

```diff
@@ -120,15 +168,85 @@ pub mod pallet {
+                Self::deposit_event(Event::StdModulePublishFailed(e.to_string()));
                 return; // can not proceed without deposit module injected
             }
-            Self::deposit_event(Event::DepositModulePublished);
+            Self::deposit_event(Event::StdModulePublished);
+
+            // Processing users modules to be published
+            ModulesToPublish::<T>::drain().for_each(|(account, id, module)| {
+                if let Err(reason) = vm.publish_module(
+                    &module,
+                    Self::native_to_move(&account).unwrap(), //FIXME: safe to unwrap?
+                    &mut UnmeteredGasMeter {},
+                ) {
+                    // report failure
+                    Self::deposit_event(Event::PublishModuleResult {
+                        publisher: account,
+                        module: id,
+                        status: ModulePublishStatus::Failure(reason.to_string()),
+                    });
+                } else {
+                    // report success
+                    Self::deposit_event(Event::PublishModuleResult {
+                        publisher: account,
+                        module: id,
+                        status: ModulePublishStatus::Suceess,
+                    });
+                }
+            });
+
+            // Executing submitted scripts
+            ScriptsToExecute::<T>::drain().for_each(|(account, id, script)| {
+                let t_args = vec![TypeTag::Signer, TypeTag::Address, TypeTag::U128];
+                if let Err(reason) =
+                    // TODO: implement after transaction merge
+                    vm.execute_script(
+                        &script,
+                        t_args,
+                        vec![
+                            &bcs::to_bytes(&Self::native_to_move(&account).unwrap()).unwrap(),
+                            &bcs::to_bytes(
+                                &Self::native_to_move(
+                                    &T::AccountId::decode(&mut [1u8; 32].as_ref()).unwrap(),
+                                )
+                                .unwrap(),
+                            )
+                            .unwrap(),
+                            &bcs::to_bytes(&123u128).unwrap(),
+                        ],
+                        &mut UnmeteredGasMeter {},
+                    )
+                {
+                    Self::deposit_event(Event::ExecuteScriptResult {
+                        publisher: account,
+                        script: id,
+                        status: ScriptExecutionStatus::Failure(reason.to_string()),
+                    });
+                } else {
+                    Self::deposit_event(Event::ExecuteScriptResult {
+                        publisher: account,
+                        script: id,
+                        status: ScriptExecutionStatus::Suceess,
+                    });
+                }
+            });
```

Comment:
We are supposed to send the delta of updates back to the main storage here somewhere? Like we discussed in Slack.